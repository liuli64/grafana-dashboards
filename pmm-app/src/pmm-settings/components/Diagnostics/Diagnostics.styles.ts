import { stylesFactory } from '@grafana/ui';
import { GrafanaTheme } from '@grafana/data';
import { css } from 'emotion';

export const getStyles = stylesFactory((theme: GrafanaTheme) => {
  const { colors }: any = theme;

  return {
    diagnosticsWrapper: css`
      align-items: flex-end;
      display: flex;
      flex: 1;
      flex-direction: column;
    `,
    diagnosticsLabel: css`
      display: flex;
      i {
        margin-left: ${theme.spacing.xs};
      }
    `,
    diagnosticsButton: css`
      background: linear-gradient(180deg, ${colors.gray15} 0%, #292c31 100%);
      border: ${theme.border.width.sm} solid ${colors.gray25};
      font-size: ${theme.typography.size.sm};
      margin-top: ${theme.spacing.md};
      &:hover {
        background: ${colors.gray15};
      }
    `
  };
});
