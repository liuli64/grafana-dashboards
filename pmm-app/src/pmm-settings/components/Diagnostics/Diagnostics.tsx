import React, { FC } from 'react';
import {
  Icon,
  LinkButton,
  Tooltip,
  useTheme
} from '@grafana/ui';
import { getStyles } from './Diagnostics.styles';
import { Messages } from '../../Settings.messages';

export const Diagnostics: FC = () => {
  const theme = useTheme();
  const styles = getStyles(theme);
  const { diagnostics: { action, label, tooltip } } = Messages;

  return (
    <div className={styles.diagnosticsWrapper}>
      <div
        className={styles.diagnosticsLabel}
        data-qa="diagnostics-label"
      >
        {label}
        <Tooltip content={tooltip}>
          <div>
            <Icon name="info-circle" />
          </div>
        </Tooltip>
      </div>
      <LinkButton
        icon={'fa fa-download' as any}
        href="/logs.zip"
        className={styles.diagnosticsButton}
        data-qa="diagnostics-button"
      >
        {action}
      </LinkButton>
    </div>
  );
};
