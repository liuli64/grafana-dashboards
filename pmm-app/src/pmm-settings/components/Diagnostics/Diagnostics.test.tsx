import React from 'react';
import { shallow } from 'enzyme';
import { Diagnostics } from './Diagnostics';

describe('Diagnostics::', () => {
  it('Renders diagnostics correctly', () => {
    const root = shallow(<Diagnostics />);

    expect(root).toMatchSnapshot();
    root.unmount();
  });
});
