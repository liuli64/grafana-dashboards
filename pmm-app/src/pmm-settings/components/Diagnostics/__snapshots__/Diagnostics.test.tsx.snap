// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diagnostics:: Renders diagnostics correctly 1`] = `
<div
  className="css-6qeqse"
>
  <div
    className="css-160v2r5"
  >
    Diagnostics
    <Tooltip
      content="You can download server logs to make the problem detection simpler. Please include this file if you are submitting a bug report."
    >
      <div>
        <Icon
          name="info-circle"
        />
      </div>
    </Tooltip>
  </div>
  <LinkButton
    className="css-n3uzy9"
    href="/logs.zip"
    icon="fa fa-download"
  >
    Download PMM Server Logs
  </LinkButton>
</div>
`;
